mono/metadata/threads.c

```diff
@@ -673,7 +673,9 @@ mono_thread_internal_set_priority (MonoInternalThread *internal, MonoThreadPrior
 
 	g_assert (internal->native_handle);
 
+	MONO_ENTER_GC_SAFE;
 	res = SetThreadPriority (internal->native_handle, priority - 2);
+	MONO_EXIT_GC_SAFE;
```

Comment:
Please capture GetLastError before MONO_EXIT_GC_SAFE. We have had problems where last error gets trashed.